datafusion/physical-plan/src/windows/mod.rs

```diff
@@ -283,7 +283,11 @@ fn create_built_in_window_expr(
                 args[1]
                     .as_any()
                     .downcast_ref::<Literal>()
-                    .unwrap()
+                    .ok_or_else(|| {
+                        DataFusionError::Execution(
+                            "Expected a signed integer literal for the second argument of nth_value".to_string(),
```

Comment:
It might be helpful to include the actual value in here. `PhysicalExpr` implements `Display`, so it should be rather easy to generate a string from that.